src/app/test_executive/gossip_consistency.ml

```diff
@@ -0,0 +1,171 @@
+    let producers = Network.block_producers network in
+    let n = List.length producers in
+    List.mapi producers ~f:(fun i node ->
+        let%map () = wait_for t (Wait_condition.node_to_initialize node) in
+        [%log info]
+          "gossip_consistency test: Block producer %d (of %d) initialized"
+          (i + 1) n ;
+        () )
+    |> Malleable_error.all_unit
+
+  let send_payments ~logger ~sender_pub_key ~receiver_pub_key ~amount ~fee
+      ~node n =
+    let open Malleable_error.Let_syntax in
+    let rec go n =
+      if n = 0 then return ()
+      else
+        let%bind () =
+          let%map () =
+            Network.Node.send_payment ~retry_on_graphql_error:false ~logger
+              ~sender_pub_key ~receiver_pub_key ~amount ~fee node
+          in
+          [%log info] "gossip_consistency test: payment #%d sent." n ;
+          ()
+        in
+        go (n - 1)
+    in
+    go n
+
+  let wait_for_payments ~logger ~dsl ~sender_pub_key ~receiver_pub_key ~amount
+      n =
+    let open Malleable_error.Let_syntax in
+    let rec go n =
+      if n = 0 then return ()
+      else
+        (* confirm payment *)
+        let%bind () =
+          let%map () =
+            wait_for dsl
+              (Wait_condition.payment_to_be_included_in_frontier
+                 ~sender_pub_key ~receiver_pub_key ~amount)
+          in
+          [%log info]
+            "gossip_consistency test: payment #%d successfully included in \
+             frontier."
+            n ;
+          ()
+        in
+        go (n - 1)
+    in
+    go n
+
+  let run network t =
+    let open Malleable_error.Let_syntax in
+    let logger = Logger.create () in
+    [%log info] "gossip_consistency test: starting..." ;
+    let%bind () = wait_for_all_to_initialize ~logger network t in
+    [%log info] "gossip_consistency test: done waiting for initializations" ;
+    let receiver_bp = Caml.List.nth (Network.block_producers network) 0 in
+    let%bind receiver_pub_key = Util.pub_key_of_node receiver_bp in
+    let sender_bp =
+      Core_kernel.List.nth_exn (Network.block_producers network) 1
+    in
+    let%bind sender_pub_key = Util.pub_key_of_node sender_bp in
+    let num_payments = 5 in
+    let fee = Currency.Fee.of_int 10_000_000 in
+    let amount = Currency.Amount.of_int 10_000_000 in
+    [%log info] "gossip_consistency test: will now send %d payments"
+      num_payments ;
+    let%bind () =
+      send_payments ~logger ~sender_pub_key ~receiver_pub_key ~node:sender_bp
+        ~fee ~amount num_payments
+    in
+    [%log info]
+      "gossip_consistency test: sending payments done. will now wait for \
+       payments" ;
+    let%bind () =
+      wait_for_payments ~logger ~dsl:t ~sender_pub_key ~receiver_pub_key
+        ~amount num_payments
+    in
+    [%log info] "gossip_consistency test: waiting for payments done" ;
+    let gossip_states = (network_state t).gossip_received in
+    let num_transactions_seen =
```

Comment:
@nholland94 as of right now, `num_transactions_seen` is always 0, not sure why.  something must be wrong with the `(network_state t).gossip_received` call

Reply:
Sounds like the `Network_state.t` is not getting updated properly. Try reading through the logic for that. The full debug logs from a `test_executive` run will probably have enough information to debug this (it tells you what events are processed from the event router and when network state gets updated).